src/IO/WriteHelpers.h

```diff
@@ -798,6 +798,36 @@ inline void writeDateTimeTextRFC1123(time_t datetime, WriteBuffer & buf, const D
     buf.write(" GMT", 4);
 }
 
+inline void writeDateTimeTextISO(time_t datetime, WriteBuffer & buf, const DateLUTImpl & utc_time_zone)
+{
+    const auto & values = utc_time_zone.getValues(datetime);
+    writeDateTimeText<'-', ':', 'T'>(
+        LocalDateTime(values.year, values.month, values.day_of_month,
+            utc_time_zone.toHour(datetime), utc_time_zone.toMinute(datetime), utc_time_zone.toSecond(datetime)), buf);
+    buf.write('Z');
+}
+
+inline void writeDateTimeTextISO(DateTime64 datetime64, UInt32 scale, WriteBuffer & buf, const DateLUTImpl & utc_time_zone)
+{
+    static constexpr UInt32 MaxScale = DecimalUtils::maxPrecision<DateTime64>();
+    scale = scale > MaxScale ? MaxScale : scale;
+
+    auto c = DecimalUtils::split(datetime64, scale);
+    const auto & values = utc_time_zone.getValues(c.whole);
+    writeDateTimeText<'-', ':', 'T'>(
+        LocalDateTime(values.year, values.month, values.day_of_month,
+            utc_time_zone.toHour(c.whole), utc_time_zone.toMinute(c.whole), utc_time_zone.toSecond(c.whole)), buf);
+    buf.write('Z');
```

Comment:
But for DateTime64 we need to write fractional part as well:
`2019-08-20T10:18:56.012345Z`

Reply:
Fixed. Thanks.